test/python/transpiler/test_unroll_3q_or_more.py

```diff
@@ -69,3 +70,15 @@ def test_decompose_conditional(self):
         for node in op_nodes:
             self.assertIn(node.name, ['h', 't', 'tdg', 'cx'])
             self.assertEqual(node.condition, (cr, 0))
+
+    def test_decompose_unitary(self):
+        """Test unrolling of unitary gate over 4qubits."""
+        qr = QuantumRegister(4, 'qr')
+        circuit = QuantumCircuit(qr)
+        unitary = random_unitary(16, seed=42)
+        circuit.unitary(unitary, [0, 1, 2, 3])
+        dag = circuit_to_dag(circuit)
+        pass_ = Unroll3qOrMore()
+        after_dag = pass_.run(dag)
+        op_nodes = after_dag.op_nodes()
+        self.assertEqual(len(op_nodes), 459)
```

Comment:
It would be better here to test properties of the UnitaryGate in the unrolled circuit (`len(op_nodes)` might change with updates to isometry, or the unroller). Maybe
```
assertTrue(isinstance(list(op_nodes)[0], UnitaryGate) and list(op_nodes)[0].params[0] == unitary.data)
```
or similar. `matrix_equal(Operator(qc).data, Operator(qc.decompose()).data)` you used in the unitary gate test would also be good.

Reply:
Good point, fixed in: https://github.com/Qiskit/qiskit-terra/pull/4243/commits/43b67a4e0bdc12d6e9cb4e38e038aecf2c334383